### PR TITLE
test(plcActivity): cover the PLC activity-log helpers

### PR DESCRIPTION
**Pull request by OPS-PIvers**
## Summary

Adds `tests/utils/plcActivity.test.ts` (27 tests) covering `utils/plcActivity.ts`, the fire-and-forget PLC activity-log module. This closes the final tracked sub-target (Priority 3, `utils/plcActivity.ts`) of the MEDIUM **"utils/ files with complex logic have no test coverage"** item in the `test-coverage` scheduled-task journal — Priority 1 (`spotifyAuth.ts`, 2026-08-03) and Priority 2 (`quizAudio.ts`, 2026-07-13) were already done, so the item now retires to Completed.

Test-only + journal update — no source/behavior changes.

## What's covered

All five exports of `utils/plcActivity.ts`:

- **`MENTION_ACTIVITY_TARGET_TYPE` / `COMMENT_ACTIVITY_TARGET_TYPE`** — the sentinel constants the feed filter and unread derivation key off.
- **`isForeignMentionEvent`** — mention privacy: to-another-member → `true`, to-self → `false`, general comment → `false`, no `targetType` → `false`, signed-out (`uid` null) → mention treated as foreign.
- **`parseActivity`** — passed `id` authoritative over a forged body `id`; minimal vs. full doc; `null` on out-of-union `type` / missing `type` / non-string `actorUid` / non-string `actorName`; `createdAt` resolution through the real `tsToMillis` (Timestamp `toMillis` / plain number / missing → `0`); non-string optional target fields dropped.
- **`deriveUnreadCount`** — null vs. numeric cursor, pending `createdAt` 0 handling in both, foreign-mention exclusion while the general comment is kept, mention-to-self counted, empty list → 0.
- **`writePlcActivityEvent`** — ref/payload shape (`id` pinned to the auto-allocated ref, `serverTimestamp()` `createdAt`, no target keys on a minimal doc), full and partial optional-target passthrough, no `logError` on success, and the fire-and-forget failure path (resolves `undefined` without throwing while `logError('writePlcActivityEvent', err, { plcId, type })` fires).

## Test approach

Stubs `firebase/firestore` (collection/doc/serverTimestamp/setDoc), `@/config/firebase` db, and `@/utils/logError`. Leaves the pure `@/utils/plc` `tsToMillis` **unmocked** so `parseActivity`'s timestamp branches run against production logic.

## Verification

- `pnpm exec vitest run tests/utils/plcActivity.test.ts` → 27 passing
- `pnpm run type-check` → clean
- `pnpm exec eslint --max-warnings 0` → clean
- `pnpm exec prettier --check` → clean

---
_Generated by [Claude Code](https://claude.ai/code/session_013FNc9gQTCCKkH5zyBWLu8h)_

### CODE REVIEW

File: tests/utils/plcActivity.test.ts [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        }),
+      ];
+      // Only the general event counts for a viewer who was not mentioned.
+      expect(deriveUnreadCount(activity, null, 'me')).toBe(1);
+    });
+
+    it('counts a mention addressed to the viewer', () => {
+      const activity = [
+        makeEvent({
+          id: 'mention-me',
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply tests/utils/plcActivity.test.ts:328]: Good catch on the uncovered branch — added a test in `3aa50fb1`.

I pinned the **actual** production behavior rather than the `.toBe(false)` in the suggestion, because `false` doesn't match the code. `isForeignMentionEvent` is `event.targetType === MENTION && event.targetId !== uid`, so with an absent `targetId` it evaluates `undefined !== uid`, which is `true` for every uid (a real one *and* `null`). A malformed mention with no `targetId` is therefore classified as **foreign (hidden)**, not surfaced. The new test asserts exactly that (`toBe(true)` for both `'me'` and `null`) and documents that it's an edge case — `buildCommentActivityEvents` always sets `targetId: mentionedUid`, so it isn't reachable on the normal write path.

Flipping this to a "safe default" (show the malformed mention to everyone) would be a behavior change to `plcActivity.ts`, which is out of scope for this test-coverage-only PR. If the team wants that semantics, it's worth a separate change to the helper (guard on `targetId == null`) — happy to file it.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by OPS-PIvers** [comment]
Thanks — all five points addressed in `ded81e46`, resolving the threads:

1. **plc-2 routing** — added `expect(mockCollection).toHaveBeenCalledWith({ __brand: 'db' }, 'plcs', 'plc-2', 'activity')`.
2. **plc-3 routing + payload fields** — added the same routing assertion for `'plc-3'` and now assert the invariant fields (`id`, `type`, `actorUid`, `actorName`, `createdAt`) alongside the target field.
3. **`vi.resetAllMocks()`** — swapped `clearAllMocks()` → `resetAllMocks()` in `beforeEach` (the mock implementations are re-established right after, so no queued `*Once` can leak).
4. **Duplicate happy-path test** — merged into the success test, now `'resolves to undefined and does not call logError on a successful write'` (file is back to 27 `it()` blocks).
5. **Journal factual errors** — test count is now genuinely 27 (matches the file after the merge); dropped `lastActiveThrottle.ts` from the remaining-untested list and noted both it (2026-07-22) and `shortLinksApi.ts` (2026-07-27) are already covered.

Verified: `vitest run` → 27 passing, `type-check`, `eslint --max-warnings 0`, and `prettier --check` all clean.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by OPS-PIvers — COMMENTED** [review]
## Automated Code Review

**Date:** 2026-08-10

### Summary
Adds a new 479-line unit suite `tests/utils/plcActivity.test.ts` covering the PLC activity-log helpers (`writePlcActivityEvent`, `parseActivity`, `isForeignMentionEvent`, `buildCommentActivityEvents`), plus routine scheduled-task doc-log updates. Test-only + docs; no production source changes.

### Code Quality
✓ No issues found. The suite mocks `@google/genai`-adjacent Firestore primitives cleanly and asserts both payload shape and `collection(...)` routing. All six reviewer threads were resolved before this review — including the `plcId`-routing false-green, the absent-`targetId` branch, and the `vi.clearAllMocks` vs `resetAllMocks` brittleness — each either fixed in `3aa50fb1` or answered with the actual production-behavior rationale.

### Widget Standards
✓ Not applicable — no widget code.

### Admin Config Standards
✓ Not applicable.

### Firestore & Security
✓ No issues found — tests exercise the write helpers against mocked Firestore; no rules or real collections touched. The suite notably pins the `actorName`-required invariant that Firestore rules depend on.

### Regression Risk
✓ Low risk. Additive test coverage; cannot affect runtime behavior.

### Missing Tests
✓ Adequate coverage — this PR *is* the coverage. The one remaining acknowledged edge (a "safe default" for malformed mentions with no `targetId`) is correctly scoped out as a helper behavior change, not a test gap.

### Overall
**Merge readiness:** Ready
Thorough, well-reasoned coverage with all review feedback addressed. Note the PR description's "27 tests" vs. the file's 28 `it()` blocks was already flagged and is cosmetic.

---
_This review was generated automatically. All findings should be verified by a human before merging._

---
_Generated by [Claude Code](https://claude.ai/code)_
